lib/main_ltp.pm

```diff
@@ -24,21 +25,25 @@ use LTP::TestInfo 'testinfo';
 use File::Basename 'basename';
 use main_common qw(load_bootloader_s390x boot_hdd_image get_ltp_tag);
 use 5.018;
-use strict;
-use warnings;
+# FIXME: Delete the "## no critic" line and uncomment following lines "use strict;" and "use warnings;"
+# use strict;
+# use warnings;
 
 our @EXPORT = 'load_kernel_tests';
 
+#  FIXME: Subroutine loadtest redefined at lib/main_ltp.pm line 35.
```

Comment:
This is probably happening because the entire file is evaluated multiple times. So there is no point commenting every function. The problem is most likely in main.pm in whatever place(s) the file is used. So I would just leave a comment at the top saying why Perl critic is disabled.

Also `use 5.018` will enable strict so there is no need to specify both. In fact it is recommended in Perl best practices to specify the version number instead of using other pragmas, but my memory is a bit fuzzy as to the exact details.

In general though this is a good idea.

Reply:
> This is probably happening because the entire file is evaluated multiple times. So there is no point commenting every function.

I just wanted to make it easier to discover what parts are the problem, but I copied things over before realizing that ever method is being redefined. I can drop the verbosity.

> The problem is most likely in main.pm in whatever place(s) the file is used. So I would just leave a comment at the top saying why Perl critic is disabled.

Just for clarification, two different things are disabled. The first is 'use warnings;' which results in all those warnings I commented in, the second is perlcritic (in regards to "strict"). The second one is only needed because it would complain that 'use warnings;' is missing :)

> Also use 5.018 will enable strict so there is no need to specify both.

Right, I'll go over the files and remove the unnecessary 'use strict;' (although I personally like the explicitness as someone who is not all that familiar with what Perl version includes which pragma by default)

What should we do about it now? Should I just create a progress ticket about fixing these warnings and re-enabling 'use warnings;' and perlcritic? I don't feel comfortable fixing those myself.

Reply:
I am not sure there is an easy way to fix it, except remove the `eval` which wraps the LTP code. However we need the `eval` to protect the rest of main.pm from errors in the LTP test runner which creates modules dynamically and in the past has been a big source of main.pm failures because we kept adding experimental features to it. Recently it has stabilised, but I have heard some people are thinking of generalising it to run non LTP tests and this will probably result in more instability for a while. 

Segregating main.pm already has an epic issue AFAIK (@okurz ?). From what we talked about I think this problem will go away if/when main.pm is rewritten.

> Right, I'll go over the files and remove the unnecessary 'use strict;' (although I personally like the explicitness as someone who is not all that familiar with what Perl version includes which pragma by default)

I agree, but unfortunately I think the order in which you use strict, warnings the version is significant. I don't know how using the version and warnings interact.

Reply:
I've removed the additional `use strict;` that I had added in files that have `use 5.018;` and dropped the additional comments in lib/main_ltp.pm